backend/sources/chat-plugin/src/main/java/co/airy/core/chat_plugin/Principal.java

```diff
@@ -13,6 +14,7 @@
 
     @Override
     public String getName() {
-        return sessionId;
+        // == conversation id
+        return UUIDV5.fromNamespaceAndName(channelId, sessionId).toString();
```

Comment:
probably over zealous _but_ we could generate in the constructor and return the cached value here